hal/src/sercom/v2/spi/reg.rs

```diff
@@ -71,6 +71,20 @@ impl<S: Sercom> Registers<S> {
         });
     }
 
+    /// Wait for the registers synchronization
+    #[inline]
+    fn is_register_synced(&mut self, synced_reg: reg_sync) -> bool {
+        #[cfg(feature = "samd20")]
+        return !self.spi().status.read().syncbusy().bit_is_set();
+        #[cfg(not(feature = "samd20"))]
+        match synced_reg {
+            reg_sync::CtrlB => !self.spi().syncbusy.read().ctrlb().bit_is_set(),
+            reg_sync::Length => !self.spi().syncbusy.read().length().bit_is_set(),
+            reg_sync::Enable => !self.spi().syncbusy.read().enable().bit_is_set(),
+            reg_sync::SwRst => !self.spi().syncbusy.read().swrst().bit_is_set(),
+        }
+    }
+
```

Comment:
I would like to look through the SAMD20 datasheet myself before we move forward with this.

Reply:
I would like to propose an alternate approach. Instead of declaring a function that takes an enum, we could do something like:
``` rust
#[cfg(not(feature = "samd20"))]
#[inline]
fn sync_ctrlb(&self) {
    while self.spi().syncbusy.read().ctrlb().bit_is_set() {}
}

#[cfg(feature = "samd20")]
#[inline]
fn sync_ctrlb(&self) {
    while self.spi().status.read().syncbusy().bit_is_set() {}
}
```

This has the advantage of not having any overhead of passing enums around, and reduces chances of making mistakes. At any rate, a shared reference to `self` is sufficient.

Also, `SYNCBUSY.LENGTH` seems to be exclusive to SAMx5x targets.